### PR TITLE
Budget App PR

### PR DESCRIPTION
**Pull request by Aseadd**
# **Hello, Good Reviewer :wave:** 

Project requirements

- Splash screen
       A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages

    The user should be able to register in the app with full name, email and password (all mandatory).
    The user can log into the app using email and password.
    If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)

When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.
- Add a new category page

The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
- Add a new transaction page
The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing requirements
Create unit and integration tests for all the most important components of your RoR application.

Screenshot from 2022-12-22 15-32-23
Screenshot from 2022-12-22 15-32-34
Screenshot from 2022-12-22 15-32-53
Screenshot from 2022-12-22 15-33-28
Screenshot from 2022-12-22 15-33-18
Screenshot from 2022-12-22 15-33-05

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,259 @@
...
+- **[key_feature_2]**
+- **[key_feature_3]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+--->
+
+<!-- LIVE DEMO -->
+
+
+## 🚀 Live Demo <a name="live-demo"></a>
```
Comment (sja-thedude) [inline README.md:99]: - [ ] Kindly make sure to also add deployed link as well, as per the requirements of the project.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @Aseadd, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Professional Readme
✔️ All linters passing
✔️ All test cases passing

## Required 
- [ ] Kindly make sure that the video presentation contains a highlight about a piece of the code the student finds interesting or is proud of making.
- [ ] _Check the comments under the review._

Note: This Capstone project is an individual activity that could lead to invalidation if someone else pushes commits too. So make sure it doesn't happen again.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (Aseadd) [reply README.md:99]: Thank you! Included!

**Review by cynthiainga — APPROVED** [review]
Hi @Aseadd, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR and README file ✔️ 
- [ ] The App is deployed correctly ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**